### ISSUE TITLE
Add generated rebar.config for restclient users

### ISSUE THREAD

**Issue by derek121** [kivra/restclient#11]
TL;DR The README does not accurately reflect how to currently use restc with rebar/rebar3 projects, and cannot be used without the generation of a rebar.config within restc (or explicitly adding restc's dependencies as dependencies of one's own project).

1) Create a rebar project with `rebar create-app appid=tester` (or the equivalent for rebar3, with the same results).

2) Create a rebar.config along the lines of

```
{deps, [
  {restc, ".*", {git, "git@github.com:kivra/restclient.git", "842a83d"}}
]}.
```

3) Add restc to applications in the tester.app.src file

4)
```
$ rebar get-deps
$ rebar compile
```

With an 18.x installation, the steps in the USAGE section fail (at the *public_key* start, due to *asn1* not being started).

Attempt based on the README:

```
1> application:start(inets).
ok
2> application:start(crypto).
{error,{already_started,crypto}}
3> application:start(public_key).
{error,{not_started,asn1}}
4> application:start(ssl).
{error,{not_started,public_key}}
5> application:ensure_all_started(tester).
{error,{hackney,{"no such file or directory",
                 "hackney.app"}}}
```

And so the restc:request(get, "https://api.github.com") call fails due to the lack of hackney.

hackney is not present because restc is an erlang.mk project, without a rebar.config, and so wasn't fetched via `rebar get-deps`.

Solution: in restc, `make rebar.config`. This generates a rebar.config, after which a rebar get-deps and rebar compile in the project of the user of restc will handle hackney and other transitive dependencies. At that point, `application:ensure_all_started(tester)` will start the required dependencies, and restc requests may be made.

erlang.mk / rebar compatibility details: http://erlang.mk/guide/compat.html#_erlang_mk_projects_as_rebar_dependencies

As the Makefile/dependencies of restc aren't going to be changing often, I'd think that the simplest solution is to add to the project the rebar.config output from `make rebar.config` - this will allow rebar projects using restc to use restc out-of-the-box, with rebar[3] picking up its dependencies. This does require that any changes to restc's Makefile dependencies be accompanied by a regeneration of the rebar.config and push, but it's transparent to users of the library. Alternatively, users could take steps such as `rebar get-deps`, `cd restc`, `make rebar.config`, `cd ..` , `rebar get-deps`, `rebar compile`, but that is not quite as easy.

PR containing a rebar.config generated from the current Makefile to follow.

**Comment by derek121** [comment]
PR: https://github.com/kivra/restclient/pull/12

**Comment by bipthelin** [comment]
Thanks.
